app/db/models/application_config.py

```diff
@@ -0,0 +1,195 @@
+    pages: Mapped[List["Page"]] = relationship(
+        "Page", order_by="Page.form_index", collection_class=ordering_list("form_index")
+    )
+    runner_publish_name = Column(db.String())
+    source_template_id = Column(UUID(as_uuid=True), nullable=True)
+
+    def __repr__(self):
+        return f"Form({self.runner_publish_name} - {self.name_in_apply['en']}, Pages: {self.pages})"
+
+
+@dataclass
+class Page(BaseModel):
+
+    form_id = Column(
+        "form_id",
+        UUID(as_uuid=True),
+        ForeignKey("form.form_id"),
+        nullable=True,  # will be null where this is a template and not linked to a form
+    )
+    page_id = Column(
+        "page_id",
+        UUID(as_uuid=True),
+        primary_key=True,
+        default=uuid.uuid4,
+    )
+    name_in_apply = Column("name_in_apply_json", JSON(none_as_null=True), nullable=False, unique=False)
+    template_name = Column("Template Name", String(), nullable=True)
+    is_template = Column("is_template", Boolean, default=False, nullable=False)
+    audit_info = Column("audit_info", JSON(none_as_null=True))
+    form_index = Column(Integer())
+    display_path = Column("display_path", String())
+    components: Mapped[List["Component"]] = relationship(
+        "Component", order_by="Component.page_index", collection_class=ordering_list("page_index")
+    )
+    source_template_id = Column(UUID(as_uuid=True), nullable=True)
+
+    def __repr__(self):
+        return f"Page(/{self.display_path} - {self.name_in_apply['en']}, Components: {self.components})"
+
+
+class Lizt(BaseModel):
+    list_id = Column(
+        UUID(as_uuid=True),
+        primary_key=True,
+        default=uuid.uuid4,
+    )
+    name = Column(String())
+    type = Column(String())
+    items = Column(JSON())
+
+
+@dataclass
+class Component(BaseModel):
+
+    component_id = Column(
+        "component_id",
+        UUID(as_uuid=True),
+        primary_key=True,
+        default=uuid.uuid4,
+    )
+    page_id = Column(
+        "page_id",
+        UUID(as_uuid=True),
+        ForeignKey("page.page_id"),
+        nullable=True,  # will be null where this is a template and not linked to a page
+    )
+    theme_id = Column(
+        UUID(as_uuid=True),
+        ForeignKey("theme.theme_id"),
+        nullable=True,  # will be null where this is a template and not linked to a theme
+    )
+    # TODO make these 2 json so we can do welsh?
+    title = Column(String())
+    hint_text = Column(String(), nullable=True)
+    options = Column(JSON(none_as_null=False))
+    type = Column(ENUM(ComponentType))
+    template_name = Column("Template Name", String(), nullable=True)
+    is_template = Column("is_template", Boolean, default=False, nullable=False)
+    audit_info = Column("audit_info", JSON(none_as_null=True))
+    page_index = Column(Integer())
+    theme_index = Column(Integer())
+    conditions = Column(JSON(none_as_null=True))
+    source_template_id = Column(UUID(as_uuid=True), nullable=True)
```

Comment:
is it worth having some sort of integrity so we cant inadvertently delete a source template (assuming this link is needed for some reason), for example

```suggestion
CREATE TABLE TemplatesTable (
    template_id INT PRIMARY KEY,
    source_template_id INT,
    -- other columns
    FOREIGN KEY (source_template_id) REFERENCES TemplatesTable(template_id) ON DELETE RESTRICT
);
```

Reply:
Possibly. I think we were keeping it so that if a template changed, we could potentially in the future find out what was created from it and update those if needed (eg. a spelling mistake). It was more kind of crystal ball gazing than a concrete need. Don't think it matters though if the source template is deleted.